inference-engine/thirdparty/clDNN/kernel_selector/core/actual_kernels/pooling/pooling_kernel_gpu_b_fs_zyx_fsv16_imad.cpp

```diff
@@ -54,12 +54,25 @@ PoolingKernelBase::DispatchData PoolingKernelGPU_b_fs_zyx_fsv16_imad::SetDefault
     auto f = out.Feature().v;
     auto b = out.Batch().v;
 
-    dispatchData.gws[0] = x;
-    dispatchData.gws[1] = y * z;
-    // we got b_fs_yx_fsv16 format, we process 16 features per workitem
-    dispatchData.gws[2] = CeilDiv(f, FEATURE_SLICE_SIZE) * b;
-    dispatchData.lws = GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo);
+    if (IsGlobalPooling(params)) {
+        y = params.inputs[0].Y().v;
+        z = params.inputs[0].Z().v;
+        dispatchData.gws[0] = x;
```

Comment:
since gws[0] is not used right now, you can set it equal to b and save some time on `b` and `f` index calc inside the kernel.